tests/unit/model/test_desired_ingestion.py

```diff
@@ -65,3 +70,73 @@ async def test_run_ingest_hooks(desired_ingestion, mocker):
             mocker.call(desired_ingestion.hook_requests[1]),
         ]
     )
+
+
+def test_relationship_is_finalized_after_source_node_added(
```

Comment:
Note to our future selves:
 Its a little sad that these tests have such a large structure to them. Maybe we need to take a look into something like [Faker](https://faker.readthedocs.io/en/master/).